Core/Core/Discussions/DiscussionHTML.swift

```diff
@@ -0,0 +1,680 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2020-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+
+enum DiscussionHTML {
+    // shortcuts to encode text for html
+    static func t(_ text: String?) -> String { CoreWebView.htmlString(text) }
+    static func s(_ text: String?) -> String { CoreWebView.jsString(text) }
+    static func s(_ styles: Styles) -> String { "'\(styles)'" }
+
+    // HTML string rendering for static content
```

Comment:
I kept the static html rendering for use in compose reply & assignment details. I could make those use the same preact rendering, but it seemed wasteful to put a bunch of js into a loadHTMLString when it's going to be static thereafter.